pkg/odo/cli/project/create.go

```diff
@@ -53,26 +53,37 @@ func (pco *ProjectCreateOptions) Validate() (err error) {
 
 // Run runs the project create command
 func (pco *ProjectCreateOptions) Run() (err error) {
+
+	successMessage := fmt.Sprintf(`Project '%s' is ready for use`, pco.projectName)
+
+	// Create the "spinner"
+	s := &log.Status{}
+
+	// If the --wait parameter has been passed, we add a spinner..
 	if pco.wait {
-		s := log.Spinner("Waiting for project to come up")
-		err = project.Create(pco.Client, pco.projectName, true)
-		if err != nil {
-			return err
-		} else {
-			s.End(true)
-			log.Successf(`Project '%s' is ready for use`, pco.projectName)
-		}
-	} else {
-		err = project.Create(pco.Client, pco.projectName, false)
-		if err != nil {
-			return err
-		}
+		s = log.Spinner("Waiting for project to come up")
+		defer s.End(false)
+	}
+
+	// Create the project & end the spinner (if there is any..)
+	err = project.Create(pco.Client, pco.projectName, pco.wait)
+	if err != nil {
+		return err
 	}
+	s.End(true)
+	log.Successf(successMessage)
 
+	// Set the current project when created if it's json output, we output a json output error
 	err = project.SetCurrent(pco.Client, pco.projectName)
 	if err != nil {
 		return err
 	}
+
+	// If -o json has been passed, let's output the approriate json output!
+	if log.IsJSON() {
+		project.MachineReadableSuccessOutput(pco.projectName, successMessage)
```

Comment:
Nit: this is a bit confusing as it feels that function is not breaking and the next line would get executed

Reply:
Got it :) Added an else statement.